allennlp/models/semantic_role_labeller.py

```diff
@@ -0,0 +1,191 @@
+from typing import Dict, Any
+
+import torch
+from torch.nn.modules.linear import Linear
+import torch.nn.functional as F
+
+from allennlp.common import Params
+from allennlp.common.tensor import arrays_to_variables, viterbi_decode
+from allennlp.data import Vocabulary
+from allennlp.data.fields import IndexField, TextField
+from allennlp.data import Instance
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder
+from allennlp.models.model import Model
+
+
+class SemanticRoleLabeller(Model):
+    """
+    This model performs semantic role labelling using BIO tags using Propbank semantic roles.
+    Specifically, it is an implmentation of `Deep Semantic Role Labelling - What works
+    and what's next <https://homes.cs.washington.edu/~luheng/files/acl2017_hllz.pdf>`_ .
+
+    This implementation is effectively a series of stacked interleaved LSTMs with highway
+    connections, applied to embedded sequences of words concatenated with a binary indicator
+    containing whether or not a word is the verbal predicate to generate predictions for in
+    the sentence. Additionally, during inference, Viterbi decoding is applied to constrain
+    the predictions to contain valid BIO sequences.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``, required
+        A Vocabulary, required in order to compute sizes for input/output projections.
+    text_field_embedder : ``TextFieldEmbedder``, required
+        Used to embed the ``tokens`` ``TextField`` we get as input to the model.
+    stacked_encoder : ``Seq2SeqEncoder``
+        The encoder (with its own internal stacking) that we will use in between embedding tokens
+        and predicting output tags.
+    """
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 stacked_encoder: Seq2SeqEncoder) -> None:
+        super(SemanticRoleLabeller, self).__init__()
+
+        self.vocab = vocab
+        self.text_field_embedder = text_field_embedder
+        self.num_classes = self.vocab.get_vocab_size("tags")
+
+        # NOTE: You must make sure that the "input_dim" of the stacked encoder
+        # in your configuration file is equal to self.text_field_embedder.output_dim + 1.
+        self.stacked_encoder = stacked_encoder
+        self.tag_projection_layer = TimeDistributed(Linear(self.stacked_encoder.get_output_dim(),
+                                                           self.num_classes))
+        # TODO(Mark): support masking once utility functions are merged.
+        self.sequence_loss = torch.nn.CrossEntropyLoss()
+
+    # pylint: disable=arguments-differ
```

Comment:
You can just put this right above the docstring, after the method declaration, indented appropriately, and then there's no need to re-enable it, because it should be scoped correctly.  (At least I'm _pretty_ sure that's how pylint works.)